### PR TITLE
Add support for GitHub Actions

### PR DESCRIPTION
**Pull request by nanashili**
# Description
This PR will add support for allowing users to see the progress of their GitHub Actions of the project that they open. Currently this is still early development and will most likely be completed this week or at latest next week.

# Related Issue
* NONE

# Known Issues

* Menu doesn't show when right clicking on the `workflow` item
* `Workflow` list disappears when selecting a `workflow` item in the list using state management

# Checklist

- [x] I read and understood the [contributing guide](https://github.com/AuroraEditor/AuroraEditor/blob/main/CONTRIBUTING.md) as well as the [code of conduct](https://github.com/AuroraEditor/AuroraEditor/blob/main/CODE_OF_CONDUCT.md)
- [x] My changes generate no new warnings
- [x] My code builds and runs on my machine
- [x] I documented my code
- [x] Review requested

# Screenshots
Screenshot 2022-09-13 at 17 25 48

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ disabled_rules:
 excluded:
   - AuroraEditorModules/.build # Where Swift Package Manager checks out dependency sources
   - AuroraEditor/Base/AuroraEditorTextView/STTextView # External package which needs to be 'fixed'
+  - AuroraEditor/Services/Model/GitHub
```
Comment (wdg) [inline .swiftlint.yml:14]: This still needs to be fixed. We should create an issue so we don’t forget it.

**Review by wdg — APPROVED** [review]

File: .swiftlint.yml (hunk 1 above)
Comment (nanashili) [reply .swiftlint.yml:14]: I'll remember, it's still a wip

File: .swiftlint.yml [hunk 2]
```diff
@@ -10,6 +10,7 @@ disabled_rules:
 # paths to ignore during linting.
 excluded:
   - AuroraEditorModules/.build # Where Swift Package Manager checks out dependency sources
+  - AuroraEditor/Base/AuroraEditorTextView/STTextView # External package which needs to be 'fixed'
```
Comment (wdg) [inline .swiftlint.yml:13]: This line can be removed too.

Comment (wdg) [inline .swiftlint.yml:13]:
```suggestion
```

**Review by wdg — CHANGES_REQUESTED** [review]

**Review by wdg — APPROVED** [review]

**Review by KaiTheRedNinja — APPROVED** [review]

**Pull request merged**